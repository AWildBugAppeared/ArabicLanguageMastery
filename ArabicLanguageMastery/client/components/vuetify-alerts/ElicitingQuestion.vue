<template>
  <base-alert color="brown lighten-1" icon="mdi-head-question">
    <template #default>
      <p>
        <slot></slot>
        <v-btn
          v-if="!showAnswer"
          class="ml-2"
          x-small
          outlined
          @click="showAnswer = true"
          >Show</v-btn
        >
      </p>
      <p v-if="showAnswer" class="answer"><slot name="answer"></slot></p>
    </template>
  </base-alert>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'ElicitingQuestion',

  data() {
    return {
      showAnswer: false,
    };
  },
});
</script>

<style lang="scss" scoped>
.answer {
  color: cyan;
  font-weight: bold;
}
</style>
