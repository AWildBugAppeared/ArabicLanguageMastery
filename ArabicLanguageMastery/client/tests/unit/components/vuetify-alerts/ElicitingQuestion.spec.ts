/* eslint-disable  @typescript-eslint/no-explicit-any */

// Imports
import Vue from 'vue';
import Vuetify from 'vuetify';

// Utilities
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';

import ElicitingQuestion from '@/components/vuetify-alerts/ElicitingQuestion.vue';

describe('ElicitingQuestion.vue', () => {
  const localVue = createLocalVue();
  let wrapper: Wrapper<Vue & { [key: string]: any }>;
  let vuetify: Vuetify;

  const mountWrapper = () => {
    wrapper = mount(ElicitingQuestion, {
      localVue,
      vuetify,
      stubs: ['BaseAlert'],
      slots: {
        answer: 'This is the answer',
        default: 'This is the question',
      },
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify({
      mocks: {
        $vuetify: {},
      },
    });
    Vue.use(Vuetify);
    document.body.setAttribute('data-app', 'true');
  });

  afterEach(() => {
    wrapper.destroy();
    jest.resetModules();
    jest.clearAllMocks();
  });

  it('should display component with answer hidden', async () => {
    mountWrapper();
    await wrapper.vm.$nextTick();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display answer when the show answer button is clicked', async () => {
    mountWrapper();
    await wrapper.vm.$nextTick();

    wrapper.get('[spec-show-answer-btn]').trigger('click');
    await wrapper.vm.$nextTick();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
