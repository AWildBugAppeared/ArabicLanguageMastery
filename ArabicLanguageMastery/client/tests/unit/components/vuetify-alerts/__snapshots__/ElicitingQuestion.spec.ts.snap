// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElicitingQuestion.vue should display answer when the show answer button is clicked 1`] = `
"<basealert-stub color=\\"brown lighten-1\\" icon=\\"mdi-head-question\\">
  <p>This is the question
    <!---->
  </p>
  <p class=\\"answer\\">This is the answer</p>
</basealert-stub>"
`;

exports[`ElicitingQuestion.vue should display component with answer hidden 1`] = `
"<basealert-stub color=\\"brown lighten-1\\" icon=\\"mdi-head-question\\">
  <p>This is the question <button type=\\"button\\" class=\\"ml-2 v-btn v-btn--outlined theme--light v-size--x-small\\" spec-show-answer-btn=\\"\\"><span class=\\"v-btn__content\\">Show</span></button></p>
  <!---->
</basealert-stub>"
`;
